cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -802,6 +806,24 @@ public WorkflowToken getWorkflowToken(ProgramId workflowId, String workflowRunId
     return runIds;
   }
 
+  /**
+   * @return true if the row key is present in the table
+   */
+  public boolean isUpgradeComplete(byte[] key) {
+    MDSKey.Builder keyBuilder = new MDSKey.Builder();
+    keyBuilder.add(key);
+    return exists(keyBuilder.build());
```

Comment:
why not compare the content of the row with `ProjectInfo.getVersion().toString()`?